### PR TITLE
Remove from vacation and compiler review group

### PR DESCRIPTION
**Pull request by jackh726**
Staying on the types review rotation, but staying off the general review queue is good for me right now.

Also move @compiler-errors since he's a full member now.

### CODE REVIEW

File: triagebot.toml [hunk 1]
```diff
@@ -608,19 +608,18 @@ cc = ["@nnethercote"]
...
 [assign.adhoc_groups]
 compiler-team = [
     "@cjgillot",
+    "@compiler-errors",
     "@petrochenkov",
     "@davidtwco",
     "@oli-obk",
     "@wesleywiser",
 ]
 compiler-team-contributors = [
```
Comment (Nilstrieb) [inline triagebot.toml:622]: maybe these two should just be merged, i dont think its very useful to have them separate

Comment (compiler-errors) [reply triagebot.toml:622]: Yeah, there's basically no reason to assign a review specifically to a compiler team member. The team group in the organization can be pinged if a FCP needs started.

Comment (jackh726) [reply triagebot.toml:622]: I thought about it, and agree. Just didn't do it in this PR so its "trivial".

**Pull request merged**